src/internal_modules/roc_core/limited_arena.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/limited_arena.h
+//! @brief Limited Arena.
+
+#ifndef ROC_CORE_LIMITED_ARENA_H_
+#define ROC_CORE_LIMITED_ARENA_H_
+
+#include "roc_core/iarena.h"
+#include "roc_core/memory_limiter.h"
+#include "roc_core/noncopyable.h"
+
+namespace roc {
+namespace core {
+
+//! Decorator around IArena to make it memory limited.
+class LimitedArena : public NonCopyable<LimitedArena>, public IArena {
```

Comment:
nit: `<LimitedArena>` is not needed here, `NonCopyable<>` would be enough.

That template argument is only used in base classes, to prevent NonCopyable to be inherited multiple times via different bases, to suppress compiler warnings. E.g. an object can inherit both RefCounted and ListNode, so RefCounted uses `NonCopyable<RefCounted>` and ListNode uses `NonCopyable<ListNode>`.

Since LimitedArena is not supposed to be inherited, this doesn't matter.